pkg/util/k8s/k8s.go

```diff
@@ -1836,3 +1836,173 @@ func GetV1beta1CRDFromFile(
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*v1.ServiceAccountList); ok {
+		for _, o := range l.Items {
+			o.Kind = "ServiceAccount"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*v1.SecretList); ok {
+		for _, o := range l.Items {
+			o.Kind = "Secret"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*v1.ConfigMapList); ok {
+		for _, o := range l.Items {
+			o.Kind = "ConfigMap"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*v1.PodList); ok {
+		for _, o := range l.Items {
+			o.Kind = "Pod"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*v1.PersistentVolumeClaimList); ok {
+		for _, o := range l.Items {
+			o.Kind = "PersistentVolumeClaim"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*appsv1.DaemonSetList); ok {
+		for _, o := range l.Items {
+			o.Kind = "DaemonSet"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*appsv1.DeploymentList); ok {
+		for _, o := range l.Items {
+			o.Kind = "Deployment"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*appsv1.StatefulSetList); ok {
+		for _, o := range l.Items {
+			o.Kind = "StatefulSet"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*rbacv1.ClusterRoleList); ok {
+		for _, o := range l.Items {
+			o.Kind = "ClusterRole"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*rbacv1.ClusterRoleBindingList); ok {
+		for _, o := range l.Items {
+			o.Kind = "ClusterRoleBinding"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*rbacv1.RoleList); ok {
+		for _, o := range l.Items {
+			o.Kind = "Role"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*rbacv1.RoleBindingList); ok {
+		for _, o := range l.Items {
+			o.Kind = "RoleBinding"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*storagev1.StorageClassList); ok {
+		for _, o := range l.Items {
+			o.Kind = "StorageClass"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*apiextensionsv1.CustomResourceDefinitionList); ok {
+		for _, o := range l.Items {
+			o.Kind = "CustomResourceDefinition"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*monitoringv1.ServiceMonitorList); ok {
+		for _, o := range l.Items {
+			o.Kind = "ServiceMonitor"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*monitoringv1.PrometheusRuleList); ok {
+		for _, o := range l.Items {
+			o.Kind = "PrometheusRule"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*monitoringv1.PrometheusList); ok {
+		for _, o := range l.Items {
+			o.Kind = "Prometheus"
+			*objs = append(*objs, o.DeepCopy())
+		}
+	} else if l, ok := list.(*monitoringv1.AlertmanagerList); ok {
+		for _, o := range l.Items {
+			o.Kind = "Alertmanager"
+			*objs = append(*objs, o.DeepCopy())
+		}
```

Comment:
Do we need to check all these objects?

Reply:
doesn't hurt IMO probably need those somedays.